tests/basic/storage_pool_test.go

```diff
@@ -2053,4 +2056,597 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool %s on node %s expansion using auto", selectedPool.Uuid, selectedNode.Name))
+		})
+
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateAndDestroy(contexts, opts)
+
+		stepLog = "destroy all the applications created before test runs"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				TearDownContext(ctx, opts)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
+	})
+})
+
+var _ = Describe("{CreateSnapshotsPoolResize}", func() {
+	/*
+		Steps : Try pool resize when ot of snapshots are created on the volume
+
+	*/
+	var testrailID = 50652
+	// Testrail Description : Try pool resize when lot of snapshots are created on the volume
+
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("CreateSnapshotsPoolResize", "Validate storage pool expansion when lots of snapshots present on the system", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	var contexts []*scheduler.Context
+	var totalSnapshotsPerVol int = 60
+
+	snapshotList := make(map[string][]string)
+	var selectedNode node.Node
+	var pickNode string
+
+	// Try pool resize when ot of snapshots are created on the volume
+	stepLog := "should get the existing storage node and expand the pool by resize-disk"
+
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("snapcreateresizepool-%d", i))...)
+		}
+		ValidateApplications(contexts)
+
+		var stNode node.Node
+
+		// Get List of Volumes presnet in the Node
+
+		stNodes := node.GetStorageNodes()
+		if len(stNodes) == 0 {
+			dash.VerifyFatal(len(stNodes) > 0, true, "No Storage Node exists !! ")
+		}
+		log.InfoD("List of Nodes present %s", stNodes)
+
+		for _, each := range contexts {
+			Volumes, err := Inst().S.GetVolumes(each)
+			log.FailOnError(err, "Listing Volumes Failed [%s]", err)
+
+			log.InfoD("Get all the details of Volumes Present")
+			for _, vol := range Volumes {
+				log.InfoD("List of Volumes to inspect %T , %s", vol, vol.ID)
+				volInspect, err := Inst().V.InspectVolume(vol.ID)
+				selectedNode := volInspect.ReplicaSets[0].Nodes
+				randomIndex := rand.Intn(len(selectedNode))
+				pickNode = selectedNode[randomIndex]
+
+				for _, n := range stNodes {
+					if n.Id == pickNode {
+						stNode = n
+					}
+				}
+				log.InfoD("Volume Inspect Details [%s]", volInspect)
+				log.FailOnError(err, "Failed to list the volumes present Err : %s", err)
```

Comment:
You are not listing volumes here, you are inspecting a volume, also same as above
```
log.FailOnError(err, "Failed to inspect volume [%s]", volInspect)
```